### PR TITLE
code for model on proportions

### LINKED ISSUE #86: addressing double-dipping with F:G in likelihood; experimenting with parametrization of p_F, p_G to improve identification; potential bias in lognormals
The current version of the model re-uses data, as it includes likelihoods with y_F, y_G, and y_F/y_G. I will explore how this impacts estimates and if it can be avoided without causing issues. Relatedly, I will explore if different parametrizations/priors for p_F and p_G can improve estimation. Lastly, the lognormal likelihoods may be biased, as the implied means on the exp scale parameters do not match what they should be (need to correct for the variance), although this may be numerically negligible.

### PR DESCRIPTION
**Pull request by njirons**
Uploading the code for the model on proportions. There are 2 versions: RW (random walk on the logit-scale proportion parameters); RE (time and location random effects on these parameters, akin to the random effects in the Facebook proportion model). Both models appear to fit the data well. My next task will be to run model selection, comparing these two via LOO-CV to see if one is substantially better than the other.

### CODE REVIEW

File: src/simulation/2_modelling/models/8_prop_model_RW.stan [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  vector<lower=0>[T * I] N; // population estimates
+  vector<lower=0>[n_FG] FG_ratio; // ratio of Instagram to Facebook user ratios
+  vector<lower=0>[T * I] p_G = p_F .* p_GF;
+  vector[T*I] mu_p_F = mu_p_F0 + to_vector(mu_p_F_t * mu_p_F_i');
+  
+  
+  // population process model
+  p[1] = softmax(append_row(logit_p[1],1));
+  N[ti_N0] = N0;
+  for (t in 2 : T) {
```
Comment (edarin) [inline src/simulation/2_modelling/models/8_prop_model_RW.stan:95]: 😍

**Review by doug-leasure — APPROVED** [review]
Very nice work :) This is a great starting point for building out an alternative model. The population process model is quite clever. With a bit more work, we may be able to use this to avoid the total population constraint in the current model (which I think may be responsible for the sampler inefficiencies). 

I committed a few changes to dive a bit deeper in the model eval:
1. a couple of plots to confirm that `p_FG` is indeed redundant with `FG_ratio`,
2. population time series plots
3. trace plots which uncovered some sampler issues that I will describe below. 

I'm happy to go ahead a merge the branch into dev, but will note a few things that we need to continue looking into...

----
There are still some sampler inefficiencies (max treedepth), but this is a significant improvement over the current model. There are a few divergent transitions--which could indicate bigger issues--but I think this is something that we can manage (the early versions of our current model also had some divergences that required some tweaking). 
```
All 4 chains finished successfully.
Mean chain execution time: 182.9 seconds.
Total execution time: 244.7 seconds.

Warning: 27 of 800 (3.0%) transitions ended with a divergence.

Warning: 123 of 800 (15.0%) transitions hit the maximum treedepth limit of 10.
```
----
The chains are not mixing for the random effects on time and location for p_F:
image
image

----
Despite the chains not mixing for `p_F` random effects, the chains for `p_F`, `p_G`, and `p_GF` are mixing well.
image

----
The time series plots look reasonable for a model without covariates.
image

----
The parameter `p_FG` is estimating observed `y_FG_ratio` very well:
image

The `p_FG` parameter is indeed redundant with the `FG_ratio` parameter:
image

**Comment by alaparicioc** [comment]
This is geat indeed! I do not have additional comments on this. I found the same as Doug. I have some questions, and they may be my mistakes. So, bear with me:

For the RW model, I found the specific problems for which there are some sampler inefficiencies are related to small sample sizes, which under my understanding may come from the use of the negtiave binomial. 

Additionally, where did you define the the priors for the overdispersion of the parameters?

In the out-of-sample leave-one-out cross-validation (log_lik_G[n] = neg_binomial_2_lpmf(y_G[n] | N[ti_G[n]] .* p_G[ti_G[n]], exp(-log_kappa_F))), did you used the log_kappa_F for Instagram? I may be using the wrong file.

In the case of the RE, I am still looking into the possibility of defining log_p_GF = log_p_G - log_p_F; instead. Just to see if helps with convergence.

**Review by alaparicioc — APPROVED** [review]

**Comment by njirons** [comment]
@alaparicioc -- sorry I missed your comments, just seeing them now. Responses below

> For the RW model, I found the specific problems for which there are some sampler inefficiencies are related to small sample sizes, which under my understanding may come from the use of the negtiave binomial.

As I mentioned in our meeting, I think the negative binomial is necessary. The posterior predictive distribution using the Poisson likelihood is not well-calibrated to the data.

> Additionally, where did you define the the priors for the overdispersion of the parameters?

I am using a flat (improper) prior on the log-scale dispersion parameters. The use of a flat prior is implied by not including a term for the dispersion parameter.

> In the out-of-sample leave-one-out cross-validation (log_lik_G[n] = neg_binomial_2_lpmf(y_G[n] | N[ti_G[n]] .* p_G[ti_G[n]], exp(-log_kappa_F))), did you used the log_kappa_F for Instagram? I may be using the wrong file.

Good catch! This is a type. It should be exp(-log_kappa_G). I will correct this.

**Pull request merged**